runtime/gc_base/GCObjectEvents.cpp

```diff
@@ -94,7 +94,7 @@ localGCReportObjectEvents(MM_EnvironmentBase *env, MM_MemorySubSpaceSemiSpace *m
 				if (extensions->objectModel.isDeadObject(objectPtr)) {
 					objectPtr = (J9Object *)((U_8 *)objectPtr + extensions->objectModel.getSizeInBytesDeadObject(objectPtr));
 				} else {
-					MM_ForwardedHeader forwardHeader(objectPtr, extensions);
+					MM_ForwardedHeader forwardHeader(objectPtr, extensions->compressObjectReferences());
```

Comment:
Perhaps it would be worthwhile to capture `extensions->compressObjectReferences()` in a local variable (outside the doubly-nested `while` loops).